File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;
import sun.jvm.hotspot.debugger.Page;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<TbBrand> findAll();

    PageResult findPage(int pageNum, int pageSize);

    void add(TbBrand brand);

    TbBrand findOne(Long id);

    void update(TbBrand brand);

    void del(Long[] ids);

    PageResult search(int pageNum, int pageSize, TbBrand brand);

    List<Map> findBrandList();

}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {

    //显示品牌数据
    this.findBrandList = function () {
        return $http.get("../brand/findBrandList")
    };

    //分页显示
    this.findPage = function (pageNum, pageSize) {
        return $http.get("../brand/findPage/" + pageNum + "/" + pageSize);
    };

    //条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post("../brand/search/" + pageNum + "/" + pageSize, searchEntity)
    };

    //添加品牌
    this.add = function (entity) {
        return $http.post("../brand/add", entity)
    };

    //修改品牌
    this.update = function (entity) {
        return $http.post("../brand/update", entity)
    };

    //在编辑窗口回显要修改的数据
    this.findOne = function (id) {
        return $http.get("../brand/findOne/" + id)
    };

    //删除品牌
    this.del = function (Ids) {
        return $http.get("../brand/del/" + Ids)
    };

    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll")
    }
});